### ISSUE TITLE
0.27.0 regression: cpptools can crash after a settings change that triggers workspace parsing

### ISSUE THREAD

**Issue by sean-mcmanus** [microsoft/vscode-cpptools#5280]
I have similar issue with cpptools 0.27.0. Here's the debug output:

```
initialized
cpptools/queryCompilerDefaults (id: 1)
Попытка получить значения по умолчанию из обнаруженного на компьютере компилятора: "".
Подходящий компилятор не найден. Укажите "compilerPath" в c_cpp_properties.json.
Попытка получить значения по умолчанию из обнаруженного на компьютере компилятора: "".
Подходящий компилятор не найден. Укажите "compilerPath" в c_cpp_properties.json.
cpptools/didChangeCppProperties
Warning: Database safety check failed. Recreating database.
[rc=0x80004005] (DB: false, Expected: true)
Служба обзора кода инициализирована
Попытка получить значения по умолчанию из обнаруженного на компьютере компилятора: "".
Подходящий компилятор не найден. Укажите "compilerPath" в c_cpp_properties.json.
Попытка получить значения по умолчанию из обнаруженного на компьютере компилятора: "".
Подходящий компилятор не найден. Укажите "compilerPath" в c_cpp_properties.json.
textDocument/didOpen: file:///c%3A/Users/%D0%9F%D0%BE%D1%81%D1%82%D0%BE%D0%B2%D0%BE%D0%B9/Documents/PlatformIO/Projects/200405-024334-esp32dev/src/sketch_apr05a.cpp
Выполняется прерывание анализа тегов C:\USERS\ПОСТОВОЙ\DOCUMENTS\PLATFORMIO\PROJECTS\200405-024334-ESP32DEV\SRC\SKETCH_APR05A.CPP и зависимостей
cpptools/activeDocumentChange: file:///c%3A/Users/%D0%9F%D0%BE%D1%81%D1%82%D0%BE%D0%B2%D0%BE%D0%B9/Documents/PlatformIO/Projects/200405-024334-esp32dev/src/sketch_apr05a.cpp
cpptools/textEditorSelectionChange
cpptools/getDocumentSymbols: file:///c%3A/Users/%D0%9F%D0%BE%D1%81%D1%82%D0%BE%D0%B2%D0%BE%D0%B9/Documents/PlatformIO/Projects/200405-024334-esp32dev/src/sketch_apr05a.cpp (id: 2)
cpptools/getCodeActions: file:///c%3A/Users/%D0%9F%D0%BE%D1%81%D1%82%D0%BE%D0%B2%D0%BE%D0%B9/Documents/PlatformIO/Projects/200405-024334-esp32dev/src/sketch_apr05a.cpp (id: 3)
Заполните кэш завершения включения.
Идет обнаружение файлов...
  Обнаружение файлов: обработано файлов — 0.
  Файлов, удаленных из базы данных: 0
Обнаружение файлов завершено.
Анализ оставшихся файлов...
  Анализ: обработано файлов — 0
Анализ оставшихся файлов завершен.
пустой цикл: повторный анализ активного документа
Проверка на наличие синтаксических ошибок: file:///c%3A/Users/%D0%9F%D0%BE%D1%81%D1%82%D0%BE%D0%B2%D0%BE%D0%B9/Documents/PlatformIO/Projects/200405-024334-esp32dev/src/sketch_apr05a.cpp
Отправка аргументов компиляции для C:\USERS\ПОСТОВОЙ\DOCUMENTS\PLATFORMIO\PROJECTS\200405-024334-ESP32DEV\SRC\SKETCH_APR05A.CPP.
  Включить: C:\Users\Постовой\Documents\PlatformIO\Projects\200405-024334-esp32dev
  stdver: ms_c++17
  intelliSenseMode: msvc-x64
cpptools/getDocumentSymbols: file:///c%3A/Users/%D0%9F%D0%BE%D1%81%D1%82%D0%BE%D0%B2%D0%BE%D0%B9/Documents/PlatformIO/Projects/200405-024334-esp32dev/src/sketch_apr05a.cpp (id: 4)
cpptools/getDocumentSymbols
Database safe to open
cpptools/getDocumentSymbols: file:///c%3A/Users/%D0%9F%D0%BE%D1%81%D1%82%D0%BE%D0%B2%D0%BE%D0%B9/Documents/PlatformIO/Projects/200405-024334-esp32dev/src/sketch_apr05a.cpp (id: 5)
cpptools/getDocumentSymbols
Постановка в очередь обновления IntelliSense для файлов в единице трансляции: C:\USERS\ПОСТОВОЙ\DOCUMENTS\PLATFORMIO\PROJECTS\200405-024334-ESP32DEV\SRC\SKETCH_APR05A.CPP
cpptools/cpptools_finishUpdateSquiggles
Число волнистых линий для ошибок: 9
Волнистые линии для ошибок будут отключены в: file:///c%3A/Users/%D0%9F%D0%BE%D1%81%D1%82%D0%BE%D0%B2%D0%BE%D0%B9/Documents/PlatformIO/Projects/200405-024334-esp32dev/src/sketch_apr05a.cpp
Время обновления IntelliSense (в секундах): 9.59
cpptools/getCodeActions: file:///c%3A/Users/%D0%9F%D0%BE%D1%81%D1%82%D0%BE%D0%B2%D0%BE%D0%B9/Documents/PlatformIO/Projects/200405-024334-esp32dev/src/sketch_apr05a.cpp (id: 6)
cpptools/didChangeCppProperties
Попытка получить значения по умолчанию из обнаруженного на компьютере компилятора: "".
Подходящий компилятор не найден. Укажите "compilerPath" в c_cpp_properties.json.
Выполняется закрытие канала связи.
Завершение работы сервера IntelliSense: C:\USERS\ПОСТОВОЙ\DOCUMENTS\PLATFORMIO\PROJECTS\200405-024334-ESP32DEV\SRC\SKETCH_APR05A.CPP
Попытка получить значения по умолчанию из обнаруженного на компьютере компилятора: "".
Подходящий компилятор не найден. Укажите "compilerPath" в c_cpp_properties.json.
Проверка на наличие синтаксических ошибок: file:///c%3A/Users/%D0%9F%D0%BE%D1%81%D1%82%D0%BE%D0%B2%D0%BE%D0%B9/Documents/PlatformIO/Projects/200405-024334-esp32dev/src/sketch_apr05a.cpp
cpptools/didChangeCppProperties
  файл анализа тегов: C:\USERS\ПОСТОВОЙ\DOCUMENTS\PLATFORMIO\PROJECTS\200405-024334-ESP32DEV\SRC\SKETCH_APR05A.CPP
cpptools/didChangeSettings
Подсистема IntelliSense: Default.
Расширенная раскраска включена.
Волнистые линии для ошибок включены, если разрешены все зависимости заголовка.
Автозавершение включено.
Исключение файла: **/.git
Исключение файла: **/.svn
Исключение файла: **/.hg
Исключение файла: **/CVS
Исключение файла: **/.DS_Store
Исключение файла: **/.vscode
Исключение поиска: **/node_modules
Исключение поиска: **/bower_components
Исключение поиска: **/*.code-search
Исключение поиска: **/.vscode
$/setTraceNotification
Заполните кэш завершения включения.
Идет обнаружение файлов...
```

The extension was installed because it was required by PlatformIO extension 1.10.0.
VS code version is 1.44.0.

_Originally posted by @me21 in https://github.com/microsoft/vscode-cpptools/issues/4833#issuecomment-611813882_

**Comment by sean-mcmanus** [comment]
@me21 Actually, "Connection to server got closed" is when the cpptools process crashes. Your crash appears to be caused by a settings change which triggers a database change -- do you know which setting you changed? So far I'm not able to get a repro (I'll try with PlatformIO later) -- are you able to figure out a way to easily repro the crash? We may be able to find a fix without a repro, since we think we know the call stack of the crash.

**Comment by elahehrashedi** [comment]
This issue is fixed in 0.28.0.
0.28.0 release: https://github.com/microsoft/vscode-cpptools/releases/tag/0.28.0
